npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/latest_version_finder.rb

```diff
@@ -58,22 +67,29 @@ def latest_version_with_no_unlock
         def lowest_security_fix_version
           return unless valid_npm_details?
 
-          versions_array =
+          secure_versions =
             if specified_dist_tag_requirement?
               [version_from_dist_tags].compact
             else possible_versions(filter_ignored: false)
             end
 
-          secure_versions = filter_vulnerable_versions(versions_array)
           secure_versions = filter_ignored_versions(secure_versions)
           secure_versions = filter_lower_versions(secure_versions)
+          secure_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(secure_versions,
+                                                                                                  security_advisories)
+
+          ensure_secure_version_available!(secure_versions)
```

Comment:
I don't think we can move this method out of `filter_ignored_versions` as the condition has now changed to raise when there are no versions left regardless of where they where filtered out.

We only want to raise `AllVersionsIgnored` when there are some versions available and the ignore conditions have filtered them all out.

I think the previous implementation should work as expected if you moved the vulnerable version filter to be before the ignored versions filter:

```
          secure_versions = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(secure_versions,
                                                                                                  security_advisories)
          secure_versions = filter_ignored_versions(secure_versions)
          secure_versions = filter_lower_versions(secure_versions)
```

Reply:
Reordered the filers like you suggested @feelepxyz which seems to do the job for this pull request 💯 

Opening a new PR to consider refactoring the filters to be order-independent. That would be outside the scope of this PR since it makes significant changes across ecosystems and we may not have test coverage for the order dependence of the filters.